blivet/devices/md.py

```diff
@@ -554,6 +556,42 @@ def preCommitFixup(self):
         if getattr(self.format, "mountpoint", None) == "/boot/efi":
             self.metadataVersion = "1.0"
 
+    def _removeStaleLVM(self):
+        """ Remove any stale LVM metadata that pre-existed in a new array's on-disk footprint.
+
+            XXX This should only be run from _postCreate.
+        """
+        import time
+        log.debug("waiting 5s for activation of stale lvm on new md array %s", self.path)
+        time.sleep(5)
+        udev.settle()
+        info = udev.get_device(self.sysfsPath)
+        fmt = getFormat(udev.device_get_format(info), device=self.path, exists=True)
+        if fmt.type != "lvmpv":
+            return
+
+        try:
+            pv_info = lvm.pvinfo(device=self.path)[self.path]
+        except errors.LVMError as e:
+            return
+
+        vg_name = None
+        vg_uuid = None
+        try:
+            vg_name = udev.device_get_vg_name(pv_info)
+            vg_uuid = udev.device_get_vg_uuid(pv_info)
+        except KeyError:
+            return
+
+        if vg_name and vg_uuid:
```

Comment:
Are really both of these required? I'd think ``vg_uuid`` could be enough. I can definitely see some race conditions where the VG uuid is available, but it's name is not (yet).

Reply:
I went back and forth on that. I considered not requiring `vg_name`. I even considered skipping the piece where I check the formatting as reported by udev because the only thing that really matters is if lvm found lv metadata. I think, though, that lvm will only find metadata if blkid/udev report the array is a pv. Anyway, you could argue to drop many of the checks and only rely on lvm's reporting and a `vg_uuid` to remove potential barriers to effectiveness. As I type this I'm leaning back in the direction of removing all of the non-essential checks.